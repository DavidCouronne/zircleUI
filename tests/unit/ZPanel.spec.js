import { shallowMount, createLocalVue } from '@vue/test-utils'
import zpanel from '@/components/z-panel.vue'
import zscroll from '@/components/z-scroll.vue'
import zslider from '@/components/z-slider.vue'
import zircle from '@/index'
const localVue = createLocalVue()
localVue.use(zircle)
const wrapper = shallowMount(zpanel, {
  propsData: {
    viewName: 'search',
    slider: true,
    progress: 80,
    color: 'success',
    imagesrc: './images/test.png',
    label: 'Test label prop'
  },
  beforeCreate () {
    this.$zircle.setView('search')
  },
  computed: {
    position: function () {
      return {X: 0, Y: 0, scale: 1, Xi: 0, Yi: 0, scalei: 1}
    },
    scrollBar: function () {
      return true
    }
  },
  slots: {
    default: '<p> lorem text </p>',
    media: '<div>youtube video</div>',
    imagesrc: './images/dummy.png',
    label: 'Test label slot',
    extension: '<z-button></z-button>'
  },
  stubs: {
    'z-scroll': zscroll,
    'z-slider': zslider
  },
  localVue
})
describe('z-panel.vue', () => {
  it('Renders required props.viewName when passed', () => {
    expect(wrapper.vm.viewName).toEqual('search')
  })
  it('Renders remaining props when passed', () => {
    expect(wrapper.vm.color).toEqual('success')
    expect(wrapper.vm.imagesrc).toEqual('./images/test.png')
    expect(wrapper.vm.label).toEqual('Test label prop')
  })
  it('Expected data.fullView processed viewName', () => {
    expect(wrapper.vm.fullView).toEqual('search--0')
  })
  it('Expected to be responsive', () => {
    expect(wrapper.vm.responsive).toEqual(true)
  })
  it('Has z-scroll component and longtext class activated because of scrollBar true', () => {
    expect(wrapper.find('.maindisc').classes()).toContain('longtext')
    expect(wrapper.find(zscroll).exists()).toBe(true)
  })
  it('Has z-slider component and progress 80%', () => {
    expect(wrapper.vm.progress).toEqual(80)
    expect(wrapper.find(zslider).exists()).toBe(true)
  })
  it('Has the expected html structure (slots, z-scroll, z-slider)', () => {
    // Note: If props.imagesrc and or props.label are present, slots.imagesrc and slots.label wont be rendered
    expect(wrapper.element).toMatchSnapshot()
  })
})
