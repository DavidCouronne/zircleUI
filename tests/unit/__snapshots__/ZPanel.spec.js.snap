// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`z-panel.vue Has the expected html structure (slots, z-scroll, z-slider) 1`] = `
<div
  class="zui main success"
  style="overflow: visible; width: 200px; height: 200px; margin: -100px 0px 0px -100px; transform: translate3d(0px, 0px, 0px) scale(1);"
  title="z-panel - search"
  type="panel"
>
  <section
    style="opacity: 1;"
  >
    <div
      class="plate"
      style="width: 260px; height: 260px; margin: -130px 0px 0px -130px;"
    />
     
    <section
      style="overflow: visible;"
    >
      <svg
        class="scroll"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="50"
          cy="50"
          r="51"
          style="transform: rotate(-45deg); stroke-dasharray: 312.1592653589793; stroke-dashoffset: 235.61944901923448; stroke-width: 3;"
        />
      </svg>
       
      <svg
        class="scroll2"
        style="transform: translate3d(70.71067811865476px, -70.71067811865474px, 0px);"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          class="handlebar"
          cx="20"
          cy="20"
          r="10"
        />
      </svg>
    </section>
     
    <svg
      class="slider"
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        r="51"
        style="transform: rotate(-90deg); stroke-dasharray: 314.1592653589793; stroke-dashoffset: 62.83185307179587; stroke-width: 3;"
      />
    </svg>
     
    <section
      class="label"
    >
      
      Test label prop
      
      <!---->
    </section>
     
    <div
      class="z-content"
    >
      <img
        height="100%"
        src="./images/test.png"
        width="100%"
      />
       
      <!---->
    </div>
     
    <div
      class="z-content maindisc longtext"
    >
      <div>
        youtube video
      </div>
       
      <div
        class="z-text"
      >
        <p>
           lorem text 
        </p>
      </div>
    </div>
     
    <div
      class="zui disc button primary"
      style="width: 30px; height: 30px; margin: -15px 0px 0px -15px; transform: translate3d(100px, 0px, 0px);"
      title="z-button"
      type="button"
    >
      <div
        class="z-pulse"
      />
       
      <!---->
       
      <div
        class="z-content"
      >
        <!---->
         
      </div>
       
      <div
        class="z-content"
      >
        <span
          class="overflow"
        />
      </div>
       
    </div>
  </section>
</div>
`;
